src/connection.js

```diff
@@ -718,8 +722,16 @@ class Connection extends EventEmitter {
     isolationLevel || (isolationLevel = this.config.options.isolationLevel);
     const transaction = new Transaction(name || '', isolationLevel);
     if (this.config.options.tdsVersion < '7_2') {
-      return this.execSqlBatch(new Request('SET TRANSACTION ISOLATION LEVEL ' + (transaction.isolationLevelToTSQL()) + ';BEGIN TRAN ' + transaction.name, callback));
+      const self = this;
+      return this.execSqlBatch(new Request('SET TRANSACTION ISOLATION LEVEL ' + (transaction.isolationLevelToTSQL()) + ';BEGIN TRAN ' + transaction.name, function() {
+        self.transactionDepth++;
+        if (self.transactionDepth === 1) {
+          self.inTransaction = true;
+        }
+        return callback.apply(this, arguments);
```

Comment:
Ah, I understand your intention for this code now.

You're right, the callback given to `new Request` gets executed with `this` bound to the request instance, and by doing `callback.apply(this, arguments)` you're wrapping the callback while forwarding this `this` value correctly. 👍 

I don't think this behaviour is necessarily right (or valid) for `beginTransactions` and friends, as the use of a `Request` object here is an implementation detail and should be of no concern to the user of the library, and we shouldn't expose this internal request object.

Also, if you check how `callback` is used a few lines below, it doesn't have a `this` binding set in that case. By using `callback.apply(null, arguments)`, we would align both code paths.